intro_to_ada.rst

```diff
@@ -3479,32 +3479,802 @@ Predefined exceptions
+
+       procedure Computational_Intensive_App is
+       begin
+          delay 0.5;
+       end Computational_Intensive_App;
+
+    end Delay_Aux_Pkg;
+
+Using this auxiliary package, we're now ready to write our time-drifting
+application:
+
+.. code-block:: ada
+
+    --% run_file: Show_Time_Drifting_Task.adb
+    --% make_flags: -gnaty
+
+    with Ada.Text_IO;   use Ada.Text_IO;
+    with Ada.Real_Time; use Ada.Real_Time;
+
+    with Delay_Aux_Pkg;
+
+    procedure Show_Time_Drifting_Task is
+       package Aux renames Delay_Aux_Pkg;
+
+       task T;
+
+       task body T is
+          Cnt   : Integer := 1;
+       begin
+          while True loop
+             delay 1.0;
+
+             Aux.Show_Elapsed_Time;
+             Aux.Computational_Intensive_App;
+
+             Put_Line ("Cycle # " & Integer'Image (Cnt));
+             Cnt  := Cnt + 1;
+          end loop;
+       end T;
+
+    begin
+       delay 5.0;
+       abort T;
+       Put_Line ("Finished time-drifting application");
+    end Show_Time_Drifting_Task;
+
+As we can see by running the application, due to the drift introduced by
+``Computational_Intensive_App``, after three iterations of the loop, we
+already have a time span of about four seconds. Using the
+:ada:`delay until` statement, however, we'll be able to avoid this time
+drift and have a regular interval of one second:
+
+.. code-block:: ada
+
+    --% run_file: Show_Cycling_Task.adb
+    --% make_flags: -gnaty
+
+    with Ada.Text_IO;   use Ada.Text_IO;
+    with Ada.Real_Time; use Ada.Real_Time;
+
+    with Delay_Aux_Pkg;
+
+    procedure Show_Cycling_Task is
+       package Aux renames Delay_Aux_Pkg;
+
+       task T;
+
+       task body T is
+          Cycle : constant Time_Span := Milliseconds (1000);
+          Next  : Time := Aux.Get_Start_Time + Cycle;
+
+          Cnt   : Integer := 1;
+       begin
+          while True loop
+             delay until Next;
+
+             Aux.Show_Elapsed_Time;
+             Aux.Computational_Intensive_App;
+
+             --  Calculate next execution time using a
+             --  cycle of one seconds
+             Next := Next + Cycle;
+
+             Put_Line ("Cycle # " & Integer'Image (Cnt));
+             Cnt  := Cnt + 1;
+          end loop;
+       end T;
+
+    begin
+       delay 5.5;
+       abort T;
+       Put_Line ("Finished cycling");
+    end Show_Cycling_Task;
+
+As we can see by running the application, the :ada:`delay until` statement
+makes sure that the ``Computational_Intensive_App`` does not affect the
+regular interval of one second between the iterations.
 
 Protected objects
 -----------------
 
-Protected objects: body
------------------------
+In situations where multiple tasks are accessing shared data, data
+corruption may happen. For example, data will be inconsistent when one
+task overwrites parts of the information that is being read by another
+task. In order to avoid this kind of problems and ensure that the
```

Comment:
It would be good to have a note (maybe in an attention/important admonition) about that vs. tasks and rendez vous, since the two mechanisms are very similar, eg. tasks can (and were before 95) be used to implement synchronized data access, but that's not practical when all you want is synchronized data.